### DISCUSSION TITLE
`\newcommand`のみの数式ブロックが空白として表示される

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#326] [要望・アイデア・不具合]
数式を利用する際に、独立した数式ブロック内で `\newcommand` を利用してコマンドを定義すると、このブロックが大きな空白として表示されます。もちろん、`\newenvironment` などでも同様の結果を得ます。おそらく、MathJax の newcommand 拡張から提供されているコマンドが対象になっていると思われます。

> [newcommand — MathJax 3.2 documentation](https://docs.mathjax.org/en/latest/input/tex/extensions/newcommand.html#newcommand-commands)
>
> The newcommand extension implements the following macros: `\def`, `\let`, `\newcommand`, `\newenvironment`, `\renewcommand`, `\renewenvironment`

## 症状

何も表示されていない空間が生じるだけなので、大きな問題にはなりませんが、あまり良い表示スタイルだとは思いません。

![image](https://user-images.githubusercontent.com/75878492/195974336-0832cfcc-7464-4f1c-98a0-9b2b9f08a6bb.png)

<details><summary>サンプルコード</summary>

```markdown
$$
\newcommand{\L}{\mathcal{L}}
$$

$\L$
```

</details>

以下の場合では、謎の棒が表示されています。

![image](https://user-images.githubusercontent.com/75878492/195974297-58ff6cc3-e7ec-4d45-90f0-422e3f28f35e.png)

<details><summary>サンプルコード</summary>

```markdown
$\newcommand{\L}{\mathcal{L}}$

$\L$
```

</details>

この問題は、表示される数式が `\newcommand` を含むブロックに入っている場合には生じません。

![image](https://user-images.githubusercontent.com/75878492/195974356-a288ab5f-58dd-4943-ba12-2e57d723808e.png)

<details><summary>サンプルコード</summary>

```markdown
$\newcommand{\L}{\mathcal{L}}\L$
```

</details>

## 悪用

ちなみに、この問題を悪用すると、パラグラフの間隔を広げることに役立ちます。数式ブロックを複数並べれば、より広い空白を得ることが出来ます。

![image](https://user-images.githubusercontent.com/75878492/195974747-4e7b321a-8d66-440e-91e3-aec40306af3b.png)

ふつうに `<br>` を使った方が簡単とは思いますが。

**Comment by tsubolabo** [comment]
[Yarakashi-Kikohshi](https://github.com/Yarakashi-Kikohshi) さん
　最終的には、Qiitaの数式処理方法の改善に期待することだと思いますが、面白そうなので、貴「宣言 即 表示方式」（添付図の例３）以外の回避方法も探してみました。

　例４のphantom付き宣言方式も使えそうです。ただし、横に半角一文字分のスペースが追加されるので、文章の途中ではなく末尾だけに使うのが良さそうです。

　なお、Qiitaほどではありませんが、自作のhtml記述のホームページでも、MathJaxのマクロを`<body>`～`</body>`間で宣言すると、行間隔が広めになってレイアウトが若干乱れることがありました。

　なお、謎の縦棒は「無駄な横スクロールバー」のできそこないのようにも見えます

<img width="951" alt="Qiita_MathJax宣言" src="https://user-images.githubusercontent.com/42261605/195981623-1fa29b61-6914-4fa4-a1ac-9a5a00364e67.png">

**Comment by tanaka-asami415** [comment]
@Yarakashi-Kikohshi @tsubolabo 
ありがとうございます！
数式周りは色々とご意見をいただいているのですが、苦戦しておりまして…😢
今回いただいた分も修正の際に参考にさせていただきます🙇

**Comment by kimkim0814** [comment]
@Yarakashi-Kikohshi 
お待たせして申し訳ありません。

>以下の場合では、謎の棒が表示されています。

上記に関しては解決しているので、ご確認頂けたらなと思います！

また、現状数式ブロック内で`\newcommand`を使用してコマンドを定義した場合も数式ブロック内に何も入力ししなかった場合も同様の大きさの空白が表示されます。
調査したところ、上記に関してはMatjJaxの仕様ですので、対応しないことをご理解頂けたらなと思います🙇

ご不便をおかけし大変申し訳ございませんが、何卒ご容赦くださいますようお願い申し上げます。

**Reply by yarakos95** [reply]
謎の棒が表示されなくなったことを確認しました 👀 
ありがとうございます 🎉 

MathJax 由来の問題はどうしようもないですね。調査していただきありがとうございます 🙇‍♂️
（どこかのタイミングで MathJax 側から修正がなされると良いなぁ）
